tests/GrapheneOS/ConnectivityCheckSettingTest/src/grapheneos/test/ConnectivityCheckSettingTest.java

```diff
@@ -0,0 +1,140 @@
+
+        ConnectivityCheckServer(String httpUrl, String httpsUrl) {
+            this.httpUrl = httpUrl;
+            this.httpsUrl = httpsUrl;
+        }
+    }
+
+    private void toggleWifi(String state) throws DeviceNotAvailableException {
+        String cmd = String.format(WIFI_CMD, state);
+        getDevice().executeShellV2Command(cmd);
+    }
+
+    @Before
+    public void setUp() throws DeviceNotAvailableException {
+        boolean isRoot = getDevice().isAdbRoot();
+        if (!isRoot) {
+            if (!getDevice().enableAdbRoot()) {
+                fail("adb root access is required");
+            }
+        }
+    }
+
+    private final ArrayList<String> parseLogcat(String regexp, int setting) throws Exception {
+        final Pattern pattern = Pattern.compile(regexp);
+        ArrayList<String> result = new ArrayList<String>();
+
+        try (InputStreamSource logSource = getDevice().getLogcat()) {
+            InputStreamReader streamReader = new InputStreamReader(logSource.createInputStream());
+            BufferedReader logReader = new BufferedReader(streamReader);
+
+            String line;
+            while ((line = logReader.readLine()) != null) {
+                Matcher matcher = pattern.matcher(line);
+                if (setting != DISABLED_SETTING) {
+                    if (matcher.find() && line.contains(LOG_TAG)) {
+                        result.add(line);
+                        if (result.toString().contains("PROBE_HTTP") && result.toString().contains("PROBE_HTTPS")) {
+                            break;
+                        }
+                    }
+                } else {
+                    if (matcher.find()) {
+                        result.add(matcher.group(0));
+                        break;
+                    }
+                }
+            }
+            return result;
+        }
+    }
+
+    private void baseTest(int setting) throws Exception {
+        toggleWifi("disabled");
+        getDevice().setProperty(SYS_PROP, Integer.toString(setting));
+        RunUtil.getDefault().sleep(1000); // 1s
+        getDevice().clearLogcat();
+        toggleWifi("enabled");
+        boolean hasConnectivity = false;
+        for (int i = 0; i < 10; ++i) {
+            RunUtil.getDefault().sleep(1000);
+            hasConnectivity = getDevice().checkConnectivity();
+            if (hasConnectivity) {
+                break;
+            }
+        }
+        RunUtil.getDefault().sleep(1000);
+        assertTrue("device has no connectivity", hasConnectivity);
+    }
+
+    @Test
+    public void testGrapheneOSServer() throws Exception {
+        baseTest(GRAPHENEOS_SERVER_SETTING);
+        ArrayList<String> parsedLogcat = parseLogcat(FILTER_REQ_REGEXP, GRAPHENEOS_SERVER_SETTING);
+        assertEquals(parsedLogcat.size(), 2);
+        assertTrue(parsedLogcat.get(0).contains(ConnectivityCheckServer.GRAPHENEOS.httpUrl) && parsedLogcat.get(1).contains(ConnectivityCheckServer.GRAPHENEOS.httpsUrl));
```

Comment:
Should use separate asserts for each condition, this simplifies finding test failure cause.